### PR TITLE
install and enable eBPF Plugin by default

### LINKED ISSUE #8650: Netdata installer needs to install specific version of `kernel-collector` repository
##### Feature idea summary
Right now `netdata-installer.sh` is always getting the latest version of `kernel-collector` repository, this can become a problem with new features and old installations.

The netdata-installer needs to do with eBPF the same we are doing with go.d.

##### Expected behavior
We will have parity between Netdata versions and kernel collector versions

* (@core-sre) Netdata-installer will download a package containing:
  - libebpf [2 versions; <4.15, 4.15+]
  - libnetdataebpf (our shared library) [single version]
  - ebpf programs [single version of processes, sockets]

* (@thiagoftsm) The collector (plugin) will make a decision at run-time to map the current kernel version onto the appropriate files.

Expected: Thursday nightly to release ebpf

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

Aside from some cleanups that ensure we use a tagged release of the [kernel-collector](https://github.com/netdata/kernel-collector) package(s) where the version is controlled via the Netdata installer; This PR also installs and Enables the eBPF plugin by default. There is now a `--disable-ebpf` for explicitly disabling the eBPF installation/plugin. And the defaul conriguration for new `netdata.conf`(s) are now:

```#!init
[plugins]
  ebpf = yes
```

##### Component Name

- area/packaging

##### Test Plan

For each Linux system:
- Spin up a Debian 10 (buster) VM
- Git clone our repo
- Checkout that PR
- Run ./netdata-installer.sh --enable-ebpf ...

##### Additional Information

- Fixes #8650 
- Fixes: #8621

### CODE REVIEW

**Comment by underhood** [comment]
not sure this should enable eBPF by default now (my impression is we need the release build ASAP, but default on it should be only on next release)

**Comment by Ferroin** [comment]
> not sure this should enable eBPF by default now (my impression is we need the release build ASAP, but default on it should be only on next release)

This isn't intended to be in the upcoming v1.21.1 release, but per @cakrit we should have eBPF enabled in v1.22.0.

**Comment by ilyam8** [comment]
> This isn't intended to be in the upcoming v1.21.1 release, but per @cakrit we should have eBPF enabled in v1.22.0.

v1.22.0 in a week, this PR should be merged asap to make ebpf testing/review possible

**Comment by Ferroin** [comment]
> > This isn't intended to be in the upcoming v1.21.1 release, but per @cakrit we should have eBPF enabled in v1.22.0.
> 
> v1.22.0 in ~a week~ the end of Apr, this PR should be merged asap to make ebpf testing/review possible

Per instructions from @amoss, I was under the impression that all the testing and review is supposed to happen here before it's merged.

**Comment by mfundul** [comment]
Debian 10 still not working:
```
 --- eBPF Kernel Collector --- 
 ERROR: Cannot detect a supported kernel on your system!
 FAILED  Incompatible kernel detected (004019000), not installing eBPF 

./netdata-installer.sh: line 1358: deferred_error: command not found
```

**Comment by thiagoftsm** [comment]
We need to update the function `get_compatible_kernel_for_ebpf()` to add the kernels 4.19, 4.19.28, 4.19.37 and 4.19.67, preferentially we should rewrite it. These kernels will be available when we merge https://github.com/netdata/kernel-collector/pull/124 .

**Comment by thiagoftsm** [comment]
@Ferroin and @prologic I finished the investigation about how we will work with the kernels and the changes merged on `kernel-collector`, you can see the details  at https://github.com/netdata/kernel-collector/issues/129 . 

Let me know any doubt that you have related to the versions, we will move from 17 different kernel versions to only 4 packages.

File: netdata-installer.sh [hunk 1]
```diff
@@ -1241,19 +1255,26 @@ get_compatible_kernel_for_ebpf() {
 
   kpkg=
 
-  if [ "${kver}" -ge 005000000 ]; then
+  # Kernel variables
+  rhver8="0080000000"
+  rhver7="0070061810"
```
Comment (mfundul) [inline netdata-installer.sh:1260]: I'd prefer if those 2 variables were called:
```
  rhver8_0="0080000000"
  rhver7_6="0070061810" 
```

File: libnetdata/ebpf/ebpf.c [hunk 2]
```diff
@@ -88,11 +89,57 @@ int get_kernel_version() {
...
     *move = '\0';
 
+    fd = snprintf(out, (size_t)size, "%s.%s.%s", major, minor, patch);
+    if (fd > size)
+        error("[EBPF]: The buffer to store kernel version is not smaller than necessary.");
+
     return ((int)(str2l(major)*65536) + (int)(str2l(minor)*256) + (int)str2l(patch));
 }
 
+int get_redhat_release()
```
Comment (mfundul) [inline libnetdata/ebpf/ebpf.c:99]: This code needs to be deduplicated from the `kernel-collector/user/process_user.c` file and should change as is described in https://github.com/netdata/kernel-collector/issues/137

File: netdata-installer.sh (hunk 1 above)
Comment (thiagoftsm) [reply netdata-installer.sh:1260]: This code will be removed @mfundul , it is not longer necessary with the new packages. I am only waiting for the final definitions on `kernel-collector` to update the code.

File: libnetdata/ebpf/ebpf.c (hunk 2 above)
Comment (thiagoftsm) [reply libnetdata/ebpf/ebpf.c:99]: I will remove it in the next commit, thank you for the suggestion!

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts.

**Comment by prologic** [comment]
@thiagoftsm and I plan to merge this PR today after some testing with a new kernel-colelctor release shortly...

**Comment by prologic** [comment]
Verified `--help` output:

```#!sh
prologic@debian:~/netdata$ ./netdata-installer.sh --help | grep -i -E ebpf
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.utf-8)

  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata                          .-.   .-
  |   '-'   '-'   '-'   '-'   '-'   installer command line options  '   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

  --disable-ebpf             Disable eBPF Kernel plugin (Default: enabled)
```

**Comment by prologic** [comment]
I guess we didn't create this PR with the intention of using the new package structure?

I now need to figure out how we should be installing this (again):

```
 --- Installing eBPF plugin ---
 Detected GLIBC
netdata-kernel-collector-glibc-v0.1.0.tar.xz: OK
 Extracting netdata-kernel-collector-glibc-v0.1.0.tar.xz ...
 Finding suitable lib directory ...
[/home/prologic/netdata]# cp -a -v /tmp/netdata-ebpf-yjfLgv/usr/lib64/libbpf_kernel.so /usr/local/lib
cp: cannot stat '/tmp/netdata-ebpf-yjfLgv/usr/lib64/libbpf_kernel.so': No such file or directory
 FAILED

[/home/prologic/netdata]# cp -a -v /tmp/netdata-ebpf-yjfLgv/libnetdata_ebpf.so /usr/libexec/netdata/plugins.d
cp: cannot stat '/tmp/netdata-ebpf-yjfLgv/libnetdata_ebpf.so': No such file or directory
 FAILED

[/home/prologic/netdata]# cp -a -v /tmp/netdata-ebpf-yjfLgv/pnetdata_ebpf_process.o /usr/libexec/netdata/plugins.d
cp: cannot stat '/tmp/netdata-ebpf-yjfLgv/pnetdata_ebpf_process.o': No such file or directory
 FAILED

[/home/prologic/netdata]# cp -a -v /tmp/netdata-ebpf-yjfLgv/rnetdata_ebpf_process.o /usr/libexec/netdata/plugins.d
cp: cannot stat '/tmp/netdata-ebpf-yjfLgv/rnetdata_ebpf_process.o': No such file or directory
 FAILED

[/home/prologic/netdata]# ln -v -f -s /usr/local/lib/libbpf_kernel.so /usr/local/lib/libbpf_kernel.so.0
'/usr/local/lib/libbpf_kernel.so.0' -> '/usr/local/lib/libbpf_kernel.so'
 OK

[/home/prologic/netdata]# ldconfig
 OK

```

cc @thiagoftsm

**Comment by thiagoftsm** [comment]
@prologic when we started to work with this PR, the new structure was not defined, we need to update it  for the new structure.

**Comment by prologic** [comment]
Okay all installer bugs/issues found are now fixed. It works on Debian 10, Ubuntu 20.04, CentOS 8, CentOS 7. The eBPF process plugin however does not actually load on Debian 10 or CentOS 7 but these are _known issues_.

As it stands right now I'm happy to merge this PR.

**Comment by mfundul** [comment]
In Ubuntu 18.04 I get:
```
2020-05-18 14:38:29: ebpf.plugin INFO  : MAIN : [EBPF_PROCESS] Cannot load library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.15.0-96-generic
7�� for the current kernel.
2020-05-18 14:38:29: ebpf.plugin INFO  : MAIN : [EBPF_PROCESS] Default shared library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.16.18 loaded with success.
[eBPF]bpf_load_program() err=22
2020-05-18 14:38:29: ebpf.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/pnetdata_ebpf_process.4.16.18.o (errno 22, Invalid argument)

```

In Debian 10 I get:
```
2020-05-18 14:36:12: ebpf.plugin INFO  : MAIN : [EBPF_PROCESS] Cannot load library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.19.0-8-amd64
 for the current kernel.
2020-05-18 14:36:12: ebpf.plugin INFO  : MAIN : [EBPF_PROCESS] Default shared library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.5.4.20 loaded with success.
[eBPF]bpf_load_program() err=22
2020-05-18 14:36:12: ebpf.plugin ERROR : MAIN : [EBPF_PROCESS] Cannot load program: /usr/libexec/netdata/plugins.d/pnetdata_ebpf_process.5.4.20.o (errno 22, Invalid argument)

```

At least Ubuntu 18.04 used to work so this appears to be a regression.

**Comment by prologic** [comment]
Re-tested on Ubuntu 20.04 to ensure that we do indeed renbale eBPF by default now. New installations now get:

```#!ini
[plugins]
  ebpf = yes
``

by default.

Things just work™ :)

**Comment by prologic** [comment]
Also tested are the following behaviors:

- On a System that does not support the correct kernel version or required Kernel config -- eBPF kernel-collector will not be installed.
- On a System where the kernel-collector libraires or eBPF programs fail to load for some reason (_Debian 10 is a KP for example_) the `ebpf.plugin` process just keep trying with a backoff but no adverse effects otherwise.

**Comment by prologic** [comment]
@mfundul 

> At least Ubuntu 18.04 used to work so this appears to be a regression.

Yeah I hope to have some integration testing at some point soon so we can actually have CI catch this.

Debian !0 however is a KP as well as CentOS 7.

**Comment by thiagoftsm** [comment]
> At least Ubuntu 18.04 used to work so this appears to be a regression.
For both, Ubuntu and Debian, we need to update the way we are detecting the kernels, it is not enough to get the version (4.19.x), we need to consider more information.
The output of `uname(2)` or `/proc/sys/kernel/osrelease` is not enough for these distributions.

**Pull request merged**